### ISSUE TITLE
[Mono X11] Error when building GodotSharp Solution

### ISSUE THREAD

**Issue by christoffersch** [godotengine/godot#20823]
**Godot version:**
v3.1.dev.mono 

**OS/device including version:**
KDE neon/Ubuntu 16.04
x86_64

**Issue description:**
Fresh clone and compile of Godot with Mono results in a bunch of warnings and a single error when _Building GodotSharp Solution_:

```
ObjectType/ARVRPositionalTracker.cs(52,39): warning CS0108: 'ARVRPositionalTracker.GetType()' hides inherited member 'object.GetType()'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/ArrayMesh.cs(40,36): warning CS0108: 'ArrayMesh.BlendShapeMode' hides inherited member 'Mesh.BlendShapeMode'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/ArrayMesh.cs(10,21): warning CS0108: 'ArrayMesh.ArrayFormat' hides inherited member 'Mesh.ArrayFormat'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/ArrayMesh.cs(23,21): warning CS0108: 'ArrayMesh.ArrayType' hides inherited member 'Mesh.ArrayType'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/CPUParticles.cs(679,22): warning CS0108: 'CPUParticles.Scale' hides inherited member 'Spatial.Scale'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/CPUParticles.cs(7,21): warning CS0108: 'CPUParticles.Flags' hides inherited member 'GeometryInstance.Flags'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/Slider.cs(68,38): warning CS0108: 'Slider.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/Expression.cs(29,74): error CS0030: Cannot convert type 'int' to 'Godot.Object' [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/PathFollow2D.cs(74,21): warning CS0108: 'PathFollow2D.Rotate' hides inherited member 'Node2D.Rotate(float)'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/LightOccluder2D.cs(28,20): warning CS0108: 'LightOccluder2D.LightMask' hides inherited member 'CanvasItem.LightMask'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

ObjectType/LineEdit.cs(139,38): warning CS0108: 'LineEdit.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [/home/christoffer/.local/share/godot/mono/solutions/11622239264844504762_2_7/GodotSharp/GodotSharp.csproj]

Done building project "GodotSharp.csproj" -- FAILED.
Done building project "GodotSharp.sln" -- FAILED.
```
...the error being:
`ObjectType/Expression.cs(29,74): error CS0030: Cannot convert type 'int' to 'Godot.Object'`

![https://i.imgur.com/ukLvYNU.png](https://i.imgur.com/ukLvYNU.png)

**Steps to reproduce:**
Following the [official docs guide](http://docs.godotengine.org/en/3.0/development/compiling/compiling_with_mono.html#example-x11) for compiling on Linux.

**Comment by aaronfranke** [comment]
What version of Mono are you using? 5.12 is required to use Godot, 4.x in Ubuntu repos does NOT work. Here's how to install the latest Mono in Ubuntu 18.04:

    sudo apt install -y apt-transport-https 
    sudo apt install -y dirmngr 
    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF 
    echo "deb https://download.mono-project.com/repo/ubuntu stable-bionic main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list 
    echo "deb https://download.mono-project.com/repo/ubuntu vs-bionic main" | sudo tee /etc/apt/sources.list.d/mono-official-vs.list 
    sudo apt install -y mono-complete
    sudo apt install -y monodevelop

**Comment by christoffersch** [comment]
@aaronfranke using mono version 5.14.0.177

Seems like this issue has been replicated on Windows: [#20864](https://github.com/godotengine/godot/issues/20864)

**Comment by KellyThomas** [comment]
@christoffersch Sorry, I obviously need to search harder before opening a new issue.

I found that this problem was fixed by #20842

**Comment by christoffersch** [comment]
@KellyThomas cool stuff, it most likely is fixed then. I will check when I get back in a few days.
